CafeNet/Controllers/LocationsController.cs

```diff
@@ -58,5 +60,23 @@ public async Task<IActionResult> Delete(long id)
 
             return Ok(new { message = "Location deleted successfully" });
         }
+
+
+        [HttpPut("{id}")]
+        [Authorize(Roles = "ADMIN")]
+        public async Task<IActionResult> Update(long id, [FromBody] UpdateLocationRequest updateLocationRequest)
+        {
+            var targetLocationId = id;
+            var currentUserRole = HttpContext.GetUserRole();
+
+            if (targetLocationId != updateLocationRequest.Id)
+                return BadRequest("ID and route does not match ID in request");
```

Comment:
You should just remove the id from route. Why is it needed anyways? Just take it from request.